graylog2-web-interface/src/components/indices/IndexMaintenanceStrategiesConfiguration.jsx

```diff
@@ -192,7 +232,24 @@ const IndexMaintenanceStrategiesConfiguration = ({
       </Row>
       <Row>
         <Col md={12}>
-          {_getConfigurationComponent(_activeSelection(), pluginExports, strategies, strategy, config, _onConfigUpdate)}
+          {shouldShowTimeBasedSizeOptimizing && retentionIsNotNoop && _getConfigurationComponent(
```

Comment:
Updated the grouped conditions. As for the `getConfigurationComponent` not sure I understand what you mean by messy, the function takes some arguments, but the logic in itself is very simple. The function is only for grouping and extracting some inline logic.

Reply:
I think it is usually nicer/cleaner to have as few arguments in a function as possible as it will be easier to read and maintain. But looking at `getConfigurationComponent` I am also not sure if there is a good way to reduce the arguments/split the method. We can also keep it this way, I don't think it is too crucial.